src/Orckestra.Composer/Providers/Localization/LocalizationExtensions.cs

```diff
@@ -51,6 +53,27 @@ public static string FormatPrice(this ILocalizationProvider localizationProvider
             return string.Format(cultureInfo, format, price);
         }
 
+        private static readonly Dictionary<string, CultureInfo> ISOCurrenciesToACultureMap =
+            CultureInfo.GetCultures(CultureTypes.SpecificCultures)
+                .Select(c => new { c, new RegionInfo(c.LCID).ISOCurrencySymbol })
+                .GroupBy(x => x.ISOCurrencySymbol)
+                .ToDictionary(g => g.Key, g => g.First().c, StringComparer.OrdinalIgnoreCase);
+
+        public static CultureInfo GetCultureByCurrencyIso(this ILocalizationProvider localizationProvider, string currencyCode)
+        {
+            ISOCurrenciesToACultureMap.TryGetValue(currencyCode, out CultureInfo cultureInfo);
```

Comment:
If the currencyCode isn't a valid cultureInfo - it is better, it is better to crash the thread earlier, than to return null and make something crash later on. 

```
if(!ISOCurrenciesToACultureMap.TryGetValue(currencyCode, out CultureInfo cultureInfo)) {
  throw new InvalidOperationException($"Not supported currency code: '{currencyCode}'");
}
```